changelog/7337.improvement.rst

```diff
@@ -0,0 +1 @@
+Warns for test functions that return non-None. Now in `pytest_pyfunc_call` after the `async_warn_and_skip` it will check for if the return is something other than None.
```

Comment:
I'd drop the second sentence here, since it's describing internal implementation details.